components/clientComponents/forms/AddressComplete/AddressComplete.tsx

```diff
@@ -0,0 +1,300 @@
+"use client";
+import { AddressCompleteChoice, AddressCompleteProps, AddressElements } from "./types";
+import { getAddressCompleteChoices, getSelectedAddress } from "./utils";
+import { Combobox, Description, Label, ManagedCombobox } from "@clientComponents/forms";
+import { useState, useEffect, useRef } from "react";
+import { useTranslation } from "@i18n/client";
+import { useField } from "formik";
+import { cn } from "@lib/utils";
+import { SpinnerIcon } from "@serverComponents/icons/SpinnerIcon";
+import { Language } from "@lib/types/form-builder-types";
+import { countries } from "@lib/managedData/countries";
+import { useFeatureFlags } from "@lib/hooks/useFeatureFlags";
+
+interface ManagedComboboxRef {
+  changeInputValue: (value: string) => void;
+}
+
+export const AddressComplete = (props: AddressCompleteProps): React.ReactElement => {
+  const { id, name, required, ariaDescribedBy } = props;
+
+  const [field, meta, helpers] = useField(props);
+
+  const { t, i18n } = useTranslation("form-builder");
+
+  //Address Complete elements
+  const [choices, setChoices] = useState<string[]>([]);
+  const [addressResultCache, setAddressResultCache] = useState<AddressCompleteChoice[]>([]); // Cache the results from the address search.
+
+  // Cache and Allow values
+  const [isReady, setIsReady] = useState(false);
+  const [allow, setAllow] = useState(false);
+  const [apiKey, setApiKey] = useState("");
+
+  const toFullAddress = (address: AddressCompleteChoice): string => {
+    return address.Text + ", " + address.Description;
+  };
+
+  const comboboxRef = useRef<ManagedComboboxRef>(null);
+
+  // Check if addressComplete is allowed.
+  const { getFlag } = useFeatureFlags();
+  const featureFlags = {
+    addressComplete: getFlag("addressComplete"),
+  };
+  useEffect(() => {
+    const checkAllowed = async () => {
+      if (featureFlags.addressComplete) {
+        setAllow(true);
+        setApiKey(process.env.NEXT_PUBLIC_ADDRESSCOMPLETE_API_KEY || "");
+      } else {
+        setApiKey("");
+      }
+
+      setIsReady(true);
+    };
+
+    if (apiKey === "") {
+      // Don't recheck.
+      checkAllowed();
+    }
+  }, [apiKey]);
+
+  //Form fillers address elements
+  const [addressObject, setAddressObject] = useState<AddressElements | null>(
+    field.value ? JSON.parse(field.value) : null
+  );
+
+  // Update the field value when the address object changes
+  useEffect(() => {
+    const newValue = addressObject ? JSON.stringify(addressObject) : "";
+    helpers.setValue(newValue);
+  }, [addressObject, helpers]);
+
+  const onAddressSearch = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!allow) {
+      return;
+    } // Abandon if addressComplete is disabled.
+
+    const query = e.target.value;
+    const responseData = await getAddressCompleteChoices(
+      apiKey,
+      query,
+      addressObject?.country || "CAN"
+    );
```

Comment:
Should we debounce this so it's not called on every keypress?

Reply:
I think that would undo some of the purpose of the control? The autocomplete has no cost associated - it's simply the result lookup when they click an address.